AndroidSDKCore/src/main/java/com/leanplum/internal/RequestOld.java

```diff
@@ -734,10 +734,21 @@ static void deleteSentRequests(int requestsCount) {
       editor.remove(Constants.Defaults.UUID_KEY);
       SharedPreferencesUtil.commitChanges(editor);
     }
-
+    // if we send less than 100% of requests, we need to reset the batch
```

Comment:
Can you please help me understand this flow. So, if the fraction is less than 100%, we will generate new BatchID, but that would be sent in a next api call? Or when will the new batch of calls be made? When the app gets killed and relaunched etc?

Reply:
At the beginning of this code block, all the requests will have the same UUID. When the fraction is 100%, everything gets sent with that UUID and the UUID gets reset. However, if we send a batch thats smaller, the same UUID can no longer be re-used and a new one must be generated for the next send.
This is all persisted so app launch is irrelevant.